complete/src/main/java/com/example/restservice/service/QueueService.java

```diff
@@ -68,14 +75,25 @@ public QueueStatusResponse getQueueStatus(String queueId) {
         .findById(queueId)
         .map(
             queue ->
-                new QueueStatusResponse(
-                    queueId,
-                    queue.getQueueName(),
-                    queue.getTokens().stream()
-                        .filter(user -> user.getStatus().equals(TokenStatus.WAITING))
-                        .count(),
-                    Long.valueOf(queue.getTokens().size()),
-                    queue.getQueueCreationTimestamp()))
+            {
+                try {
+                    CustomQuestions customQuestions = queue.getCustomQuestions() != null
```

Comment:
You have duplicate code starting with line +122

Reply:
Also you can do with flatMap and return 
`
Option.ofNullable(queue.getCustomQuestions()).map(this::deserializeQuestions).map(this::createQueueStatusResponse)
`

Reply:
Also why do we store json? We can handle this using table relationship and we can avoid this complicated logic. Plus it adds a lot of computational overhead.

Reply:
Agreed. If you want to redo this as a table, with just textual questions and answers, we can do that. But we should think first about how we would display the user input along with each token in the admin page. Maybe some collapsable expansion panel would do the job.

That's the reason why I have been delaying this feature for a long time, till all the easy features are done and I get some real customers.